sqldelight-gradle-plugin/src/test/fixtures/well-formed-selects/expected/com/sample/TeamModel.java

```diff
@@ -138,4 +135,19 @@ public T won_cup(boolean won_cup) {
       return (T) this;
     }
   }
+
+  final class Factory<T extends TeamModel> {
```

Comment:
It's too hard for me to reason about in the diff. I'll look post-merge when it's easier to poke around.